coriolis_openstack_utils/actions.py

```diff
@@ -753,3 +755,157 @@ def execute_operations(self):
+        super(SecurityGroupCreationAction, self).print_operations()
+        secgroup_name = self.get_new_secgroup_name()
+        LOG.info(
+            "Create new destination secgroup named '%s' with source's "
+            "security rules." % (secgroup_name))
+
+    def get_new_secgroup_name(self):
+        return self.secgroup_name_format % {
+            "original": self.payload["source_name"]}
+
+    def create_secgroup_body(self, description):
+        return {"name": self.get_new_secgroup_name(),
+                "tenant_id": self.get_destination_tenant_id(),
+                "description": description}
+
+    def execute_operations(self):
+        super(SecurityGroupCreationAction, self).print_operations()
+        dest_secgroup_name = self.get_new_secgroup_name()
+        dest_tenant_id = self.get_destination_tenant_id()
+        done = self.check_already_done()
+        if done["done"]:
+            LOG.info(
+                "Security Group named '%s' already exists.",
+                dest_secgroup_name)
+            return done["result"]
+
+        LOG.info("Creating destination security group with name '%s'" %
+                 dest_secgroup_name)
+
+        description = (self.NEW_SECGROUP_DESCRIPTION %
+                       self.payload['source_name'])
+
+        body = self.create_secgroup_body(description)
+        dest_secgroup_id = security_groups.create_security_group(
+            self._destination_client, dest_tenant_id, body)
+
+        LOG.info("Adding source %s rules to destination security group '%s'" %
+                 (self.payload['source_name'], dest_secgroup_name))
+
+        src_secgroup_name = self.payload.get('source_name')
+        src_tenant_id = self.get_source_tenant_id()
+
+        source_secgroup_list = security_groups.list_security_groups(
+            self._source_client, src_tenant_id,
+            secgroup_name=src_secgroup_name)
+
+        if not source_secgroup_list:
+            raise Exception("Source security group named %s in tenant %s "
+                            "not found!" % (src_secgroup_name, src_tenant_id))
+        else:
+            source_secgroup = source_secgroup_list.pop()
+
+        src_rules = source_secgroup['security_group_rules']
+        new_dest_rules = (
+            security_groups.create_destination_security_group_rules(
+                dest_secgroup_id, src_rules))
+
+        prev_dest_rules = security_groups.list_security_groups(
+            self._destination_client, dest_tenant_id,
+            secgroup_name=dest_secgroup_name).pop()['security_group_rules']
+
+        # Removing conflicting rules
+        dest_rules = []
+        for new_rule in new_dest_rules:
+            if new_rule not in dest_rules:
+                skip_rule = False
+                for prev_rule in prev_dest_rules:
+                    if security_groups.check_rule_similarity(
+                            new_rule, prev_rule):
+                        skip_rule = True
```

Comment:
I think it's absolutely worth at least a `LOG.debug()` whenever the tools make any "rash decisions" such as deciding to skip a secgroup rule. (also paste the body of it secgroup rule for reference)